rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -177,31 +279,52 @@ private static FlowNode getStageExecStartNode(List<FlowNode> allNodesSorted, int
      * @return The name of the node on which the supplied FlowNode executed.
      */
     public static String getExecNodeName(FlowNode flowNode) {
-        String execNodeName = FlowNodeExecutorNameCache.getExecNodeName(flowNode);
+        if (flowNode == null) {
```

Comment:
Agreed, either annotate the parameter as `@CheckForNull` (and specify in Javadoc what it means for the node to be null), or mark it `@Nonnull` and do not runtime check and make sure callers comply.